packages/ui/components/form/inputs/TextField.tsx

```diff
@@ -170,13 +217,8 @@ export const InputField = forwardRef<HTMLInputElement, InputFieldProps>(function
               className
             )}
             {...passThrough}
-            {...(type == "search" && {
-              onChange: (e) => {
-                setInputValue(e.target.value);
-                props.onChange && props.onChange(e);
-              },
-              value: inputValue,
-            })}
+            onChange={handleChange}
+            value={type === "number" ? inputValue : undefined}
```

Comment:
`type === 'text'` (or any other beside number) will always be undefined. This is a bug.